// @tanstack/vue-form/dist/cjs/useField.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const formCore = require("@tanstack/form-core");
const vueStore = require("@tanstack/vue-store");
const vue = require("vue");
function useField(opts) {
  const fieldApi = (() => {
    const api = new formCore.FieldApi({
      ...opts,
      form: opts.form,
      name: opts.name
    });
    const extendedApi = api;
    extendedApi.Field = Field;
    return extendedApi;
  })();
  const fieldState = vueStore.useStore(fieldApi.store, (state) => state);
  let cleanup;
  vue.onMounted(() => {
    cleanup = fieldApi.mount();
  });
  vue.onUnmounted(() => {
    cleanup();
  });
  vue.watch(
    () => opts,
    () => {
      fieldApi.update({ ...opts, form: opts.form });
    }
  );
  return { api: fieldApi, state: fieldState };
}
const Field = vue.defineComponent(
  (fieldOptions, context) => {
    const fieldApi = useField({ ...fieldOptions, ...context.attrs });
    return () => context.slots.default({
      field: fieldApi.api,
      state: fieldApi.state.value
    });
  },
  { name: "Field", inheritAttrs: false }
);
exports.Field = Field;
exports.useField = useField;
