// .ignored/@tanstack/vue-form/dist/esm/useField.js
import { FieldApi } from "@tanstack/form-core";
import { useStore } from "@tanstack/vue-store";
import { defineComponent, onMounted, onUnmounted, watch } from "vue";
function useField(opts) {
  const fieldApi = (() => {
    const api = new FieldApi({
      ...opts,
      form: opts.form,
      name: opts.name
    });
    const extendedApi = api;
    extendedApi.Field = Field;
    return extendedApi;
  })();
  const fieldState = useStore(fieldApi.store, (state) => state);
  let cleanup;
  onMounted(() => {
    cleanup = fieldApi.mount();
  });
  onUnmounted(() => {
    cleanup();
  });
  watch(
    () => opts,
    () => {
      fieldApi.update({ ...opts, form: opts.form });
    }
  );
  return { api: fieldApi, state: fieldState };
}
const Field = defineComponent(
  (fieldOptions, context) => {
    const fieldApi = useField({ ...fieldOptions, ...context.attrs });
    return () => context.slots.default({
      field: fieldApi.api,
      state: fieldApi.state.value
    });
  },
  { name: "Field", inheritAttrs: false }
);
export {
  Field,
  useField
};
